hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/FixedFileTrailer.java

```diff
@@ -624,19 +624,22 @@ private String getHBase1CompatibleName(final String comparator) {
     return comparatorKlass;
   }
 
-  public static CellComparator createComparator(
-      String comparatorClassName) throws IOException {
+  static CellComparator createComparator(String comparatorClassName) throws IOException {
+    if (comparatorClassName.equals(CellComparatorImpl.COMPARATOR.getClass().getName())) {
+      return CellComparatorImpl.COMPARATOR;
+    } else if (comparatorClassName.equals(
+        CellComparatorImpl.META_COMPARATOR.getClass().getName())) {
+      return CellComparatorImpl.META_COMPARATOR;
+    }
```

Comment:
With out this also (with below way of get class based on the name) it will work? U just added a short circuit here? Or u fixing some issue here with below getComparatorClass(comparatorClassName)?

Reply:
I think it is to avoid the reflections call if possible

Reply:
Just to avoid the reflection in near all cases.